lms/djangoapps/courseware/grades.py

```diff
@@ -126,9 +127,18 @@ def grade(student, request, course, keep_raw_scores=False):
     """
     Wraps "_grade" with the manual_transaction context manager just in case
     there are unanticipated errors.
+    Send a signal to update the minimum grade requirement status.
     """
     with manual_transaction():
-        return _grade(student, request, course, keep_raw_scores)
+        grade_summary = _grade(student, request, course, keep_raw_scores)
+        MIN_GRADE_REQUIREMENT_STATUS.send_robust(
+            sender=None,
+            request=request,
```

Comment:
Few failed test cases are 
test_submissions_api_overrides_scores (lms.djangoapps.courseware.tests.test_submitting_problems.TestCourseGrader)

ERROR: test_weighted_exam (lms.djangoapps.courseware.tests.test_submitting_problems.TestCourseGrader)
 
ERROR: test_weighted_homework (lms.djangoapps.courseware.tests.test_submitting_problems.TestCourseGrader)

The error is     
````
lms/djangoapps/courseware/grades.py line 136 in grade
      request=request.user.username,
   AttributeError: 'WSGIRequest' object has no attribute 'user'
````